Source/OCMock/OCMock.h

```diff
@@ -42,8 +42,13 @@
 ({ \
     _OCMSilenceWarnings( \
         [OCMMacroState beginStubMacro]; \
-        invocation; \
-        [OCMMacroState endStubMacro]; \
+        OCMStubRecorder *recorder = nil; \
+        @try{ \
+            invocation; \
+        }@finally{ \
+            recorder = [OCMMacroState endStubMacro]; \
+        } \
+        recorder; \
     ); \
```

Comment:
Because invocation can throw an exception which would cause +endStubMacro to not get called.  Now that the previously-global->now-per-thread macro state is held strongly by the thread dictionary, we can no longer count on the autorelease scope magically causing -dealloc to happen, and thus -dealloc can't do the same cleanup that +endStubMacro did.